pkg/config/runtime.go

```diff
@@ -105,19 +104,15 @@ func (r *Runtime) resolveRules(bag attribute.Bag, aspectSet AspectSet, rules []*
 
 // combined returns a Combined config given an aspect config
 func (r *Runtime) combined(aa *pb.Aspect, aspectSet AspectSet) *Combined {
-	if !aspectSet[aa.GetKind()] {
-		glog.V(3).Infof("Aspect Rejected %v not is set (%v)", aa.GetKind(), aspectSet)
+	if !aspectSet[aa.Kind] {
+		glog.V(3).Infof("Aspect rejected: %v not in set [%v]", aa.Kind, aspectSet)
 		return nil
 	}
 
 	var adp *pb.Adapter
 	// find matching adapter
 	// assume that config references are correct
-	if aa.GetAdapter() != "" {
-		adp = r.adapterByName[aa.GetAdapter()]
-	} else {
-		adp = r.adapterByKind[aa.GetKind()][0]
-	}
+	adp = r.adapterByName[aa.Adapter]
```

Comment:
This doesn't seem like a good idea in general. This makes the order in which adapters are declared to be meaningful, which would be surprising to the operator.

This is also an example of design decisions made to try to make hand-editing of the config state easier by avoiding the need to type in a line of config text.

Let's make this explicit, it has to be specified.